test/tests/functional/pbs_hook_execjob_end.py

```diff
@@ -457,3 +457,33 @@ def test_qrun_arrayjob_on_mom_breakdown(self):
         self.server.expect(JOB, {'job_state': 'B'}, jid)
         subjid_1 = j.create_subjob_id(jid, 1)
         self.common_steps(subjid_1, self.mom)
+
+    def test_mom_restart(self):
+        """
+        Test to restart mom while execjob_end hook is running
+        """
+        hook_name = "end_hook"
+        attr = {'event': 'execjob_end', 'enabled': 'True', 'alarm': '40'}
+        hook_body = ("import pbs\n"
```

Comment:
instead of doing like this, I suggest do like as below: https://github.com/PBSPro/pbspro/blob/d189e05a39c57060dc6f59ba7360f791b8a19885/test/tests/functional/pbs_hook_management.py#L51-L59